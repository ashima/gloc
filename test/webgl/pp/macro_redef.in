#define A
#define A

#define B 1
#define B 1

#define C 1
//#define C 2 // this is the only illegal one

void foo() {
     __LINE__;
     __FILE__;
}

#if GL_ES
#define __FILE__ 6
#else
#define __FILE__ foo
#endif
#define __LINE__ 1000

void bar() {
     __LINE__;
     __FILE__;
}

#undef __FILE__
#undef __LINE__

#if GL_ES
#define __FILE__ 13
#else
#define __FILE__ bar
#endif
#define __LINE__ 6000

void main() {
     __LINE__;
     __FILE__;
}
