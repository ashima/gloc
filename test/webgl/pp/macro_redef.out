File 0, line 16, col 8 - 16:
"__FILE__" is a reserved macro and may not be redefined

File 0, line 20, col 8 - 16:
"__LINE__" is a reserved macro and may not be redefined

File 0, line 27, col 7 - 15:
"__FILE__" is a reserved macro and may not be undefined

File 0, line 28, col 7 - 15:
"__LINE__" is a reserved macro and may not be undefined

File 0, line 31, col 8 - 16:
"__FILE__" is a reserved macro and may not be redefined

File 0, line 35, col 8 - 16:
"__LINE__" is a reserved macro and may not be redefined

Fatal: unrecoverable preprocessor error (3)
